src/DI/WebpackExtension.php

```diff
@@ -110,4 +114,44 @@ public function beforeCompile()
 		}
 	}
 
+
+	private function setupAssetResolver(array $config): ServiceDefinition
+	{
+		$builder = $this->getContainerBuilder();
+
+		$assetResolver = $builder->addDefinition($this->prefix('assetResolver'))
+			->setClass(AssetResolver\AssetResolverInterface::class);
+
+		if ($config['manifest']['name'] !== NULL) {
+			Validators::assertField($config['manifest'], 'name', 'string:1..');
+			$loader = $builder->addDefinition($this->prefix('manifestLoader'))
+				->setClass(ManifestLoader::class);
+
+			$args = [
+				$config['manifest']['name'],
+				$loader,
+			];
+
+			if ($config['manifest']['optimize']) {
+				$devServerInstance = new DevServer($config['devServer']['enabled'], $config['devServer']['url'], new Client());
+				$directoryProviderInstance = new BuildDirectoryProvider($config['build']['directory'], $devServerInstance);
+				$loaderInstance = new ManifestLoader($directoryProviderInstance);
+
+				$manifestCache = $loaderInstance->loadManifest($config['manifest']['name']);
+				$args[] = $manifestCache;
```

Comment:
This parameter constructiom can be replaced with something like StaticAssetResolver, it does not need to depend on filename and manifest loader.